sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,634 @@
+        `config` to compare the weight shapes of the passed pre-trained model
+        to those in the config. If there are weight size mismatches, the model
+        will reinitialize new weights to match the weight shapes inside the `config`.
+        The `y` argument will then be fit to fine-tune the model. In the case where
+        there are no newly initialized weights (i.e the config weight shapes match
+        the pretrained model weight shapes), it will behave the same as the "full"
+        strategy where only the `model_path` is passed in.
+
+        - If `fit_strategy` is set to "zero-shot", requires only the `model_path`
+        parameter. It will load the model via the `fit` function with the argument
+        `model_path` and ignore any passed `y`.
+    validation_split : float, optional, default = 0.2
+        Fraction of the data to use for validation.
+    config : dict, optional, default = {}
+        A config dict specifying parameters to initialize an full
+        PatchTST model. Missing parameters in the config will be automatically
+        replaced by their default values. See the PatchTSTConfig config on
+        huggingface for more details.
+        Note: if `prediction_length` is passed as in larger than the passed `fh`
+        in the `fit` function, the `prediction_length` will be used to train the
+        model. If `prediction_length` is passed as in smaller than the passed
+        `fh` in the `fit` function, the passed `fh` will be used to train the
+        model.
+    training_args : dict, optional, default = None
+        Training arguments to use for the model. If this is passed,
+        the remaining applicable training arguments will be ignored
+    compute_metrics : list or function, default = None
+        List of metrics or function to use during training
+    callbacks: list or function, default = None
+        List of callbacks or callback function to use during training
+
+    References
+    ----------
+    [1] A Time Series is Worth 64 Words: Long-term Forecasting with Transformers
+        Yuqi Nie, Nam H. Nguyen, Phanwadee Sinthong, Jayant Kalagnanam
+        Paper: https://arxiv.org/abs/2211.14730
+    [2] HuggingFace PatchTST Page:
+        https://huggingface.co/docs/transformers/en/model_doc/patchtst
+
+    Examples
+    --------
+    >>> #Example with a new model initialized from config only
+    >>> from sktime.forecasting.hf_patchtst_forecaster import HFPatchTSTForecaster
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> forecaster = HFPatchTSTForecaster(
+    ... config = {
+    ...     "patch_length": 1,
+    ...      "context_length": 2,
+    ...      "patch_stride": 1,
+    ...      "d_model": 64,
+    ...      "num_attention_heads": 2,
+    ...      "ffn_dim": 32,
+    ...      "head_dropout": 0.3,
+    ...    },
+    ...    training_args = {
+    ...         "output_dir":"/PatchTST/",
```

Comment:
```suggestion
    ...         "output_dir":"test_output",
```
also do this at all other places